### PR TITLE
boards: arm: add ubx_bmd340eval_nrf52840

### PR DESCRIPTION
**Pull request by ghost**
Add support for BMD-340-EVAL board from u-blox AG

Signed-off-by: Bob Recny <bob.recny@u-blox.com>

### CODE REVIEW

**Comment by brec-u-blox** [comment]
I just returned from holiday and thought I'd check progress. It looks like the automated checks passed.

File: boards/arm/ubx_bmd340eval_nrf52840/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+Power (J5)
+
++-------+--------------+-------------------------+
+| PIN # | Signal Name  | BMD-34x Functions       |
++=======+==============+=========================+
+| 1     | VSHLD        | N/A                     |
++-------+--------------+-------------------------+
+| 2     | VSHLD        | N/A                     |
++-------+--------------+-------------------------+
+| 3     | RESET        | P0.21 / RESET           |
```
Comment (anangl) [inline boards/arm/ubx_bmd340eval_nrf52840/doc/index.rst:298]: In nRF52840, the nRESET function is on P0.18.

**Review by anangl — APPROVED** [review]
Looks good. Only one minor issue spotted in the documentation .

File: boards/arm/ubx_bmd340eval_nrf52840/doc/index.rst (hunk 1 above)
Comment (brec-u-blox) [reply boards/arm/ubx_bmd340eval_nrf52840/doc/index.rst]: Fixed - thanks for the catch!

**Review by ioannisg — APPROVED** [review]

**Pull request merged**